### DISCUSSION TITLE
Transforming JSX for SSR

### DISCUSSION THREAD

**Discussion by yonathan06** [solidjs/solid#742] [Q&A] [answered]
I'm having trouble using `babel-preset-solid` for transforming JSX for SSR code (maybe I should use a different transformer)
For example:

App.tsx:

```
import { JSX } from 'solid-js';
interface AppProps {
  children: JSX.Element
}
export default function App(props: AppProps) {
  return <div>{props.children}</div>;
}
```

transformed to App.js:
```
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = App;

var _web = require("solid-js/web");

const _tmpl$ = (0, _web.template)(`<div></div>`, 2);

function App(props) {
  return (() => {
    const _el$ = _tmpl$.cloneNode(true);

    (0, _web.insert)(_el$, () => props.children);
    return _el$;
  })();
}
```

The problem is, I'm using the comp in the server, using `renderToStringAsync`:
```
import { renderToStringAsync } from "solid-js/web";
import App from "./App";

fastify.get("/", async (req, res) => {
    const app = await res.send(
      await renderToStringAsync(() => <App/>)
    );
  });
```

The problem is `babel-preset-solid` is transforming the JSX code using `_web.template(...)` function which is only used for browser code, but it is not present on the server code, so I get the error `template` is not a function, which is right because it is undefined

**Comment by yonathan06** [comment, marked as answer]
Answering to my self:
Found the option for `babel-preset-solid` to transform for SSR in babel config:

```
module.exports = {
  presets: [
   ...
    ["babel-preset-solid", { generate: "ssr", hydratable: true  }],
  ],
  ...
};
```

**Reply by ryansolid** [reply]
Sorry for the poor docs about that.. They are sort of in a small section on the website and should be better reflected in the readme. https://www.solidjs.com/guide#server-side-rendering
